Library/ViewModels/PostCampaignCheckoutViewModel.swift

```diff
@@ -130,6 +136,35 @@ public class PostCampaignCheckoutViewModel: PostCampaignCheckoutViewModelType,
         AppEnvironment.current.environmentType.stripePublishableKey
       )
     }
+
+    // MARK: Validate Checkout Details On Submit
+
+    let stripeCardIdAndPaymentIntent = self.creditCardSelectedSignal.signal
+
+    let validateCheckout = Signal.combineLatest(checkoutId, stripeCardIdAndPaymentIntent)
+      .takeWhen(self.submitButtonTappedProperty.signal)
+      .switchMap { checkoutId, stripeCardIdAndPaymentIntent in
+        let (stripeCardId, paymentIntent) = stripeCardIdAndPaymentIntent
+
+        return AppEnvironment.current.apiService
+          .validateCheckout(
+            checkoutId: checkoutId,
+            paymentSourceId: stripeCardId,
+            paymentIntentClientSecret: paymentIntent
+          )
+          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+          .materialize()
+      }
+
+    self.validateCheckoutSuccess = stripeCardIdAndPaymentIntent
+      .takeWhen(validateCheckout.values().ignoreValues())
+      .map { stripeCardIdAndPaymentIntent in
+        let (_, paymentIntent) = stripeCardIdAndPaymentIntent
+        return paymentIntent
+      }
+
+    self.showErrorBannerWithMessage = validateCheckout.errors()
+      .map { _ in Strings.Something_went_wrong_please_try_again() }
```

Comment:
Nit: Will we show any of the errors to the user, or always this one?

Reply:
Unfortunately, we don't have any neat server-side error messaging. I was told generic messaging is fine for now 😢